src/views/login/components/login.js

```diff
@@ -4,16 +4,38 @@ import './login.scss';
 export default class Login extends Component {
 
     static propTypes = {
+        usernameChanged: PropTypes.func.isRequired,
+        passwordChanged: PropTypes.func.isRequired,
+        login: PropTypes.func.isRequired,
+
+        usernameField: PropTypes.string.isRequired,
+        passwordField: PropTypes.string.isRequired,
+        history: PropTypes.object.isRequired,
     };
 
+    handleUsernameChanged = (e) => {
+        const { usernameChanged } = this.props;
+        usernameChanged(e.target.value);
```

Comment:
These two functions are basically doing nothing because there is no `dispatch`, right?  And if we wanted them to do something it would be good style to include that in `user-action-creators.js` instead of here, right?